### DISCUSSION TITLE
Failed to update and no logs

### DISCUSSION THREAD

**Discussion by MuhammedKilinc60** [pocketbase/pocketbase#2099] [Q&A] [answered]
Hello my pocketbase has 2 problems 
1: Nothing is logged anymore 
2: Cannot update my collection 

```js
const d = JSON.stringify({
        "user": pb.authStore.model.id,
        "viewstring": JSON.stringify(e.source),
        "hiddens": JSON.stringify(e.target)
      })
      console.log(d)
      const res = await pb.collection("views").update(filteredvws[active].id, d)
      ```
![image](https://user-images.githubusercontent.com/125026145/226307473-42fbedcb-b806-49fc-b6c9-365642d6efbe.png)

**Comment by ganigeorgiev** [comment]
The `update()` bodyParams argument should be either a plain JS object or [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData).
Or in other words, remove the `JSON.stringify` and pass directly the object as body and the SDK will take care for normalizing it.

**Reply by MuhammedKilinc60** [reply]
thanks for your reply but i get the same response

**Reply by ganigeorgiev** [reply]
I'm not sure how to reproduce it. Please elaborate with more information about your collection structure and what the API rules are.

You can also start PocketBase with `--debug` for a more detailed request logs and/or check the logs in the "Admin UI > Logs".

**Reply by MuhammedKilinc60** [reply]
There are no logs even i can fetch there are no logs for fetching either

**Reply by ganigeorgiev** [reply]
@MuhammedKilinc60 Without any information about your collections structure, I'm not sure how to help. Please provide more details.

**Reply by MuhammedKilinc60** [reply]
this is the structure

```json
[
    {
        "id": "_pb_users_auth_",
        "name": "users",
        "type": "auth",
        "system": false,
        "schema": [],
        "listRule": "id = @request.auth.id",
        "viewRule": "id = @request.auth.id",
        "createRule": "",
        "updateRule": "id = @request.auth.id",
        "deleteRule": "id = @request.auth.id",
        "options": {
            "allowEmailAuth": true,
            "allowOAuth2Auth": true,
            "allowUsernameAuth": true,
            "exceptEmailDomains": null,
            "manageRule": null,
            "minPasswordLength": 8,
            "onlyEmailDomains": null,
            "requireEmail": false
        }
    },
    {
        "id": "g54c2pmdq5yav9t",
        "name": "views",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "rhofxy9z",
                "name": "user",
                "type": "relation",
                "system": false,
                "required": true,
                "unique": false,
                "options": {
                    "collectionId": "_pb_users_auth_",
                    "cascadeDelete": true,
                    "minSelect": null,
                    "maxSelect": 1,
                    "displayFields": [
                        "id",
                        "username"
                    ]
                }
            },
            {
                "id": "ii7iqoym",
                "name": "viewstring",
                "type": "json",
                "system": false,
                "required": true,
                "unique": false,
                "options": {}
            },
            {
                "id": "aasoa6xw",
                "name": "hiddens",
                "type": "json",
                "system": false,
                "required": false,
                "unique": false,
                "options": {}
            }
        ],
        "listRule": "",
        "viewRule": null,
        "createRule": "",
        "updateRule": "",
        "deleteRule": null,
        "options": {}
    }
]

```

**Reply by MuhammedKilinc60** [reply]
![image](https://user-images.githubusercontent.com/125026145/226311521-548eaac1-ae10-492c-85a5-c87f22b04687.png)

**Comment by MuhammedKilinc60** [comment, marked as answer]
I guess it was because I started pb without sudo

**Reply by ganigeorgiev** [reply]
`sudo` is usually not required unless you've used it the first time when you've started PocketBase.

This is because the first time when you start PocketBase with sudo it will create the `pb_data` and its db file with the root account, meaning that any other user not part of the root group will not be allowed to modify the `pb_data`.

If you don't want to use `sudo`, you can update the permissions of your existing directory with `chmod` and/or change the owner to your regular user with `chown`.
